vms/proposervm/vm.go

```diff
@@ -353,17 +386,38 @@ func (vm *VM) SetPreference(ctx context.Context, preferred ids.ID) error {
 	// custom VMs. Until the P-chain is modified to target a specific block
 	// time, ProposerMinBlockDelay can be configured in the subnet config.
 	minDelay = math.Max(minDelay, vm.MinBlkDelay)
+	nextStartTime := parentTimestamp.Add(minDelay)
 
-	preferredTime := blk.Timestamp()
-	nextStartTime := preferredTime.Add(minDelay)
-	vm.Scheduler.SetBuildBlockTime(nextStartTime)
+	return nextStartTime, nil
+}
 
-	vm.ctx.Log.Debug("set preference",
-		zap.Stringer("blkID", blk.ID()),
-		zap.Time("blockTimestamp", preferredTime),
-		zap.Time("nextStartTime", nextStartTime),
+func (vm *VM) resetPostDurangoScheduler(
+	ctx context.Context,
+	blkHeight uint64,
+	pChainHeight uint64,
+	parentTimestamp time.Time,
+	currentTime time.Time,
+) (time.Time, error) {
+	nextStartTime, err := vm.Windower.MinDelayForProposer(
+		ctx,
+		blkHeight,
+		pChainHeight,
+		parentTimestamp,
+		vm.ctx.NodeID,
+		currentTime,
 	)
-	return nil
+	if errors.Is(err, proposer.ErrNoSlotsScheduledInNextFuture) {
+		// This node is not scheduled to propose in any nearby slots. To avoid
+		// costly iteration, stop iterating early and only continue iterating if
+		// needed.
+		return currentTime.Add(proposer.MaxLookAheadWindow), nil
+	}
+	if err != nil {
+		vm.ctx.Log.Debug("failed to calculate min delay for proposer",
+			zap.Error(err),
+		)
+	}
+	return nextStartTime, err
```

Comment:
I think we need to take into account `MinBlkDelay` here.

Reply:
Done